manifests/facts.pp

```diff
@@ -41,10 +41,10 @@
   if $facts["os"]["family"] == "windows" {
     # on windows task scheduler prevent dupes already so no need to handle the PID here
     scheduled_task{"mcollective_facts_yaml_refresh":
-      ensure               => $cron_ensure,
-      command              => $rubypath,
-      arguments            => "'${scriptpath}' -o '${factspath}'",
-      trigger              => {
+      ensure    => $cron_ensure,
+      command   => $rubypath,
+      arguments => "'${scriptpath}' -o '${factspath}' --sleep '${fqdn_rand($refresh_interval, 'facts cronjob')}'",
```

Comment:
hmm, except perhaps the bit where the program will sleep in memory for up to interval seconds which is wasteful?

Screwit, adjust the cronjob and schedule time - if it breaks on windows its probably a bug  in Puppet and not really our problem, either works for me